draft/real-world-phoenix-lets-deploy-using-kubernetes.md

```diff
@@ -0,0 +1,643 @@
+
+```json
+resource "digitalocean_tag" "kubernetes-cl01" {
+  name = "kubernetes-cl01"
+}
+
+resource "digitalocean_kubernetes_cluster" "cl01" {
+  name    = "cl01"
+  region  = "ams3"
+  version = "1.16.2-do.1"
+
+  node_pool {
+    name       = "default"
+    size       = "s-1vcpu-2gb"
+    node_count = 2
+    tags       = ["${digitalocean_tag.kubernetes-cl01.id}"]
+  }
+}
+```
+
+Now you can get your cluster setup by issuing the following commands in the folder you
+created:
+
+```bash
+tarraform init
+terraform apply
+```
+
+This will take about 5 minutes and then you'll have a fresh kubernetes cluster
+up and running. Now that was easy!
+
+### Install Traefik
+
+Kubernetes needs an ingress controller to route traffik from the outside world
+to the services that are running inside the cluster as these services are not
+exposed to the outside world, which is a good thing! Nginx is often used, but I went with Traefik, a very nice alternative that has a lot of
+nice functinality out of the box, like automatic ssl certificates using Let's Encrypt
+and auto discovery of services running in the cluster.
+
+Create a file called `traefik-values.yml` in the root of your kubernetes config
+directory and add the following content. If you manage dns settings with
+digitalocean, you can comment out those parts as well. This will provide
+automatic ssl certificate generation. Very cool! If not you can just leave it
+like this. It'll work, but without ssl enabled.
+
+```yaml
+
+image: traefik
+dashboard:
+  enabled: true
+  domain: domain.example.com                     # put a (sub)domain here where you want to access the traefik dashboard
+serviceType: LoadBalancer
+rbac:
+  enabled: true
+# ssl:
+#   enabled: true                                # Enables SSL
+#   enforced: true                               # Redirects HTTP to HTTPS
+# acme:
+#   enabled: true                                # Enables Let's Encrypt certificates
+#   staging: false                               # Use Lets Encrypt staging area for this example. For production purposes set this to false
+#   email: info@drumusician.com                  # Email address that Let's Encrypt uses to notify about certificate expiry etc.
+#   challengeType: "dns-01"
+#   dnsProvider:
+#     name:  digitalocean                        # This is why you need your domain to be under Digital Ocean control
+#     digitalocean:
+#       DO_AUTH_TOKEN: $DIGITALOCEAN_TOKEN
+#   domains:
+#     enabled: true
+#     domainsList:
+#       - main: domain.example.com               # domain that belongs to this certificate
+```
+
+Now we can go ahead and install traefik in our cluster:
+
+Install helm:
+```bash
+brew install kubernetes-helm
+```
+
+Pull in our fresh cluster configuration locally (otherwise helm will
+fail):
+```bash
+doctl kubernetes cluster kubeconfig save cl01
+```
+
+And install traefik:
+
+```bash
+helm install traefik --values traefik-values.yml stable/traefik
```

Comment:
I had to add the Helm stable charts repo first (see https://helm.sh/docs/intro/quickstart/):

```bash
helm repo add stable https://kubernetes-charts.storage.googleapis.com/
```